PWGHF/D2H/TableProducer/dataCreatorDV0Reduced.cxx

```diff
@@ -0,0 +1,184 @@
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file dataCreatorDV0Reduced.cxx
+/// \brief Creation of D-V0 pairs
+///
+/// \author Luca Aglietta <luca.aglietta@cern.ch>, UniTO Turin
+
+#include <cmath>
+#include <map>
+
+#include "CommonConstants/PhysicsConstants.h"
+#include "DCAFitter/DCAFitterN.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/O2DatabasePDGPlugin.h"
+#include "Framework/runDataProcessing.h"
+#include "ReconstructionDataFormats/DCA.h"
+
+#include "Common/Core/trackUtilities.h"
+#include "Common/DataModel/CollisionAssociationTables.h"
+
+#include "PWGLF/DataModel/LFStrangenessTables.h"
+
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "PWGHF/Utils/utilsBfieldCCDB.h"
+#include "PWGHF/D2H/DataModel/ReducedDataModel.h"
+
+using namespace o2;
+using namespace o2::analysis;
+using namespace o2::aod;
+using namespace o2::constants::physics;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+// event types
+enum Event : uint8_t {
+  Processed = 0,
+  NoDV0Selected,
+  DV0Selected,
+  kNEvent
+};
+
+/// Creation of D-V0 pairs
+struct HfDataCreatorDV0Reduced {
+  // Produces AOD tables to store track information
+  Produces<aod::StraCollisions> lfReducedCollision;  // Defined in PWGLF/DataModel/LFStrangenessTables.h
+  Produces<aod::HfOrigColCounts> hfCollisionCounter; // Defined in PWGHF/D2H/DataModel/ReducedDataModel.h
+  Produces<aod::HfRedVzeros> hfCandV0;               // Defined in PWGHF/D2H/DataModel/ReducedDataModel.h
+  Produces<aod::HfRedD> hfCandD;                     // Defined in PWGHF/D2H/DataModel/ReducedDataModel.h
+
+  // selection
+  Configurable<int> selectionFlagD{"selectionFlagD", 7, "Selection Flag for D"};
+
+  HfHelper hfHelper;
+
+  // O2DatabasePDG service
+  Service<o2::framework::O2DatabasePDG> pdg;
+
+  bool isHfCandResoConfigFilled = false;
+
+  using TracksPidAll = soa::Join<aod::pidTPCFullEl, aod::pidTPCFullMu, aod::pidTPCFullPi, aod::pidTPCFullKa, aod::pidTPCFullPr,
+                                 aod::pidTOFFullEl, aod::pidTOFFullMu, aod::pidTOFFullPi, aod::pidTOFFullKa, aod::pidTOFFullPr>;
+  using TracksPIDWithSel = soa::Join<aod::TracksWCovDcaExtra, TracksPidAll, aod::TrackSelection>;
+  using TracksPIDWithSelAndMc = soa::Join<TracksPIDWithSel, aod::McTrackLabels>;
+  using CandsDFiltered = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>>;
```

Comment:
The Ds1 decays into D*, while the Ds2* decays into D+. Depending which channel you want to implement here I would be consistent (either you rename the histogram Ds2Star, or you use the D* instead of the D+ if you aim at implementing the Ds1, as I would suggest indeed)